### PR TITLE
Deprecate and remove usages of rb_define_dummy_encoding()

### PR DESCRIPTION
**Pull request by eregon**

### CODE REVIEW

File: ext/win32ole/win32ole.c [hunk 1]
```diff
@@ -687,10 +687,19 @@ ole_cp2encoding(UINT cp)
         }
     }
 
-    enc_name = rb_sprintf("CP%d", cp);
-    idx = rb_enc_find_index(enc_cstr = StringValueCStr(enc_name));
+    switch(cp) {
+      case 20932:
+        enc_cstr = "eucJP-ms";
```
Comment (eregon) [inline ext/win32ole/win32ole.c:692]: Maybe we should handle other mappings from `ole_encoding2cp()`? See https://github.com/ruby/ruby/blob/2aa8edaec701ca2d76e7be7a9b792b69574ac869/ext/win32ole/win32ole.c#L491

**Comment by eregon** [comment]
@nurse @nobu What do you think of these changes?

I found that these changes are enough to pass all tests, and remove all usages of `rb_define_dummy_encoding()`.
So I think  `rb_define_dummy_encoding()` is never actually used or necessary in practice.

File: ext/win32ole/win32ole.c (hunk 1 above)
Comment (eregon) [inline ext/win32ole/win32ole.c:692]: Is mapping CR20932 to `eucJP-ms` correct? It seems so from the mapping in ole_encoding2cp and the comments in tests (`WIN32OLE.codepage = 20932 # MS EUC-JP`)

**Comment by nurse** [comment]
> @nurse @nobu What do you think of these changes?
> 
> I found that these changes are enough to pass all tests, and remove all usages of `rb_define_dummy_encoding()`. So I think `rb_define_dummy_encoding()` is never actually used or necessary in practice.

As far as about nkf, I'll reject the change. This issue should be handled as meeting log says
> * rb_enc_from_index's lock
>     * naruse: I don't know a use case which creates infinite dummy encodings though I know about ten dummies.
>     * ko1: Some tests generates but they are tests
>     * ko1: if we can use fixed sized array, it improves the performance
>     * naruse: At this time we have 103 encodings. 256 must be sufficient (if there's a use case, they should repot to us)
>     * Conclusion: Change the encoding table into fixed array whose size is 256

**Comment by eregon** [comment]
@nurse Could you give me an example of these ten dummies?
The fact that no test reveals the usage of those this sounds to me like it's a non-issue in practice.
Especially for `nkf`, it seems effectively dead code, could you show me a counter-example?

And if we need them, why not defining them in Ruby already so they don't need to defined dynamically?

**Comment by eregon** [comment]
For NKF, AFAIK all encodings listed in https://docs.ruby-lang.org/en/master/NKF.html are already built-in in Ruby, so I don't see the need for dummy encodings.

**Comment by eregon** [comment]
https://github.com/ruby/ruby/pull/6935 was merged instead.
I would prefer to deprecate and remove `rb_define_dummy_encoding()`, but I respect the meeting decision to instead limit to 256 encodings.

**Pull request closed without merging**